kv/kvserver/raftlog: TestIterator failed

kv/kvserver/raftlog.TestIterator [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12726416?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12726416?buildTab=artifacts#/) on release-23.2 @ [0456c0d4e4dca39bde1b4bc146bf7158db3c0f30](https://github.com/cockroachdb/cockroach/commits/0456c0d4e4dca39bde1b4bc146bf7158db3c0f30):

Fatal error:

```
panic: logging already active; first use:
```
Stack: 

```
goroutine 926 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/log.setActive()
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:437 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(_, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:263 +0x50
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1dd0dc0, 0xc0006bef30}, 0x4, 0x1, 0x3, 0x0?, {0x1a71665, 0x2}, {0xc00052c400, 0x1, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:109 +0x5a5
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1dd0dc0?, 0xc0006bef30?}, 0x0?}, {0x1a71665?, 0x41eb67?}, {0xc00052c400?, 0x17a6700?, 0x1?})
	github.com/cockroachdb/cockroach/pkg/storage/pebble.go:726 +0x56
github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x0?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:668 +0x9e
github.com/cockroachdb/pebble.TeeEventListener.func13({0xc00017d200?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:743 +0x42
github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc00017d200)
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x322
created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0xcc
 [recovered]
	panic: logging already active; first use:
goroutine 926 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/log.setActive()
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:437 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(_, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:263 +0x50
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1dd0dc0, 0xc0006bef30}, 0x4, 0x1, 0x3, 0x0?, {0x1a71665, 0x2}, {0xc00052c400, 0x1, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:109 +0x5a5
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1dd0dc0?, 0xc0006bef30?}, 0x0?}, {0x1a71665?, 0x41eb67?}, {0xc00052c400?, 0x17a6700?, 0x1?})
	github.com/cockroachdb/cockroach/pkg/storage/pebble.go:726 +0x56
github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x0?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:668 +0x9e
github.com/cockroachdb/pebble.TeeEventListener.func13({0xc00017d200?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:743 +0x42
github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc00017d200)
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x322
created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0xcc
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestIterator
I231117 13:52:31.349992 926 3@pebble/event.go:743  [s?,pebble] 1  [JOB 4] all initial table stats loaded
--- FAIL: TestIterator (0.00s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIterator.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33608

Passes 1k stress runs locally. Kicking off 100k runs with:


```
dev test pkg/kv/kvserver/raftlog -f TestIterator -v --stress -- --define -gotags=bazel,gss,deadlock
```


<details><summary>full stacks from failure</summary>


```
13:52:31     --- FAIL: TestIterator (0.00s)
I231117 13:52:31.349992 926 3@pebble/event.go:743  [s?,pebble] 1  [JOB 4] all initial table stats loaded
13:52:31     panic: logging already active; first use:
13:52:31     goroutine 926 [running]:
13:52:31     runtime/debug.Stack()
13:52:31       GOROOT/src/runtime/debug/stack.go:24 +0x65
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.setActive()
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/clog.go:437 +0xa7
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(_, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/clog.go:263 +0x50
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1dd0dc0, 0xc0006bef30}, 0x4, 0x1, 0x3, 0x0?, {0x1a71665, 0x2}, {0xc00052c400, 0x1, ...})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/channels.go:109 +0x5a5
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
13:52:31       github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
13:52:31     github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1dd0dc0?, 0xc0006bef30?}, 0x0?}, {0x1a71665?, 0x41eb67?}, {0xc00052c400?, 0x17a6700?, 0x1?})
13:52:31       github.com/cockroachdb/cockroach/pkg/storage/pebble.go:726 +0x56
13:52:31     github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x0?})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:668 +0x9e
13:52:31     github.com/cockroachdb/pebble.TeeEventListener.func13({0xc00017d200?})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:743 +0x42
13:52:31     github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc00017d200)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x322
13:52:31     created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0xcc
13:52:31      [recovered]
13:52:31       panic: logging already active; first use:
13:52:31     goroutine 926 [running]:
13:52:31     runtime/debug.Stack()
13:52:31       GOROOT/src/runtime/debug/stack.go:24 +0x65
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.setActive()
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/clog.go:437 +0xa7
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(_, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/clog.go:263 +0x50
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1dd0dc0, 0xc0006bef30}, 0x4, 0x1, 0x3, 0x0?, {0x1a71665, 0x2}, {0xc00052c400, 0x1, ...})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/channels.go:109 +0x5a5
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
13:52:31       github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
13:52:31     github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1dd0dc0?, 0xc0006bef30?}, 0x0?}, {0x1a71665?, 0x41eb67?}, {0xc00052c400?, 0x17a6700?, 0x1?})
13:52:31       github.com/cockroachdb/cockroach/pkg/storage/pebble.go:726 +0x56
13:52:31     github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x0?})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:668 +0x9e
13:52:31     github.com/cockroachdb/pebble.TeeEventListener.func13({0xc00017d200?})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:743 +0x42
13:52:31     github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc00017d200)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x322
13:52:31     created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0xcc
13:52:31     
13:52:31     
13:52:31     goroutine 985 [running]:
13:52:31     testing.tRunner.func1.2({0x19348a0, 0xc0004a67f8})
13:52:31       GOROOT/src/testing/testing.go:1526 +0x24e
13:52:31     testing.tRunner.func1()
13:52:31       GOROOT/src/testing/testing.go:1529 +0x39f
13:52:31     panic({0x19348a0, 0xc0004a67f8})
13:52:31       GOROOT/src/runtime/panic.go:884 +0x213
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log/logcrash.ReportOrPanic({0x1dd0d50, 0xc000136000}, 0x0, {0x1aae5fd?, 0xc00084f3e0?}, {0xc000210520?, 0x0?, 0x0?})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:417 +0x1b4
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig({{0xc0002100f0, 0xc00084f260, 0xc00084f280, 0xc00084f3d8, 0xc00084f29d, {0x1, 0xc0001d5b50, 0x0, 0xc00084f249, 0xc00084f24a, ...}}, ...})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/flags.go:96 +0x10d
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.newLogScope.func1(0xc0006d1c50, {0x1ddb808, 0xc000728680}, 0x32?)
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:165 +0x1b8
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.newLogScope({0x1ddb808, 0xc000728680}, 0xb3?)
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:172 +0x3df
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.ScopeWithoutShowLogs({0x1ddb808, 0xc000728680})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:101 +0x38
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.Scope({0x1ddb808, 0xc000728680})
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:80 +0x4e
13:52:31     github.com/cockroachdb/cockroach/pkg/kv/kvserver/raftlog.TestIterator(0xc000728680)
13:52:31       github.com/cockroachdb/cockroach/pkg/kv/kvserver/raftlog/iter_test.go:169 +0x54
13:52:31     testing.tRunner(0xc000728680, 0x1ba52b8)
13:52:31       GOROOT/src/testing/testing.go:1576 +0x10b
13:52:31     created by testing.(*T).Run
13:52:31       GOROOT/src/testing/testing.go:1629 +0x3ea
13:52:31     
13:52:31     goroutine 1 [chan receive]:
13:52:31     testing.(*T).Run(0xc00057a820, {0x1a7ef9b?, 0x600a05?}, 0x1ba52b8)
13:52:31       GOROOT/src/testing/testing.go:1630 +0x405
13:52:31     testing.runTests.func1(0x2bd1dc0?)
13:52:31       GOROOT/src/testing/testing.go:2036 +0x45
13:52:31     testing.tRunner(0xc00057a820, 0xc00049fba0)
13:52:31       GOROOT/src/testing/testing.go:1576 +0x10b
13:52:31     testing.runTests(0xc000529e00?, {0x29e9560, 0x3, 0x3}, {0xc000378648?, 0x100c00049fc28?, 0x2bd0a00?})
13:52:31       GOROOT/src/testing/testing.go:2034 +0x489
13:52:31     testing.(*M).Run(0xc000529e00)
13:52:31       GOROOT/src/testing/testing.go:1906 +0x63a
13:52:31     main.main()
13:52:31       main/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/raftlog/raftlog_test_/testmain.go:140 +0x468
13:52:31     
13:52:31     goroutine 52 [syscall]:
13:52:31     os/signal.signal_recv()
13:52:31       GOROOT/src/runtime/sigqueue.go:152 +0x2f
13:52:31     os/signal.loop()
13:52:31       GOROOT/src/os/signal/signal_unix.go:23 +0x19
13:52:31     created by os/signal.Notify.func1.1
13:52:31       GOROOT/src/os/signal/signal.go:151 +0x2a
13:52:31     
13:52:31     goroutine 82 [chan receive]:
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:117 +0x59
13:52:31     created by github.com/cockroachdb/cockroach/pkg/util/log.init.5
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:83 +0x25
13:52:31     
13:52:31     goroutine 83 [chan receive]:
13:52:31     github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:141 +0x96
13:52:31     created by github.com/cockroachdb/cockroach/pkg/util/log.init.5
13:52:31       github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:84 +0x31
13:52:31     
13:52:31     goroutine 868 [sync.Cond.Wait]:
13:52:31     sync.runtime_notifyListWait(0xc0008560a0, 0x0)
13:52:31       GOROOT/src/runtime/sema.go:527 +0x14c
13:52:31     sync.(*Cond).Wait(0x0?)
13:52:31       GOROOT/src/sync/cond.go:70 +0x8c
13:52:31     github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
13:52:31       github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:204
13:52:31     github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0xc000856000, {0x1823840, 0x2c0b430})
13:52:31       github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:436 +0x692
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368700?, 0x0?, 0x0?}}, 0xc000309fc0)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble/record.NewLogWriter.func2()
13:52:31       github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:355 +0x5c
13:52:31     created by github.com/cockroachdb/pebble/record.NewLogWriter
13:52:31       github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:354 +0x416
13:52:31     
13:52:31     goroutine 94 [select]:
13:52:31     github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
13:52:31       github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:171 +0xdc
13:52:31     created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker
13:52:31       github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:166 +0x5d
13:52:31     
13:52:31     goroutine 851 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c1e0})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc00063afa8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 852 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c180})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc000639fa8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0xc00063f680?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 850 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*cleanupManager).mainLoop(0xc00022a0a0)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:155 +0xe7
13:52:31     github.com/cockroachdb/pebble.openCleanupManager.func1.1({0x1dd0dc0, 0xc0004ba030})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:86 +0x1d
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368800?, 0x0?, 0x0?}}, 0xc000691fc0)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.openCleanupManager.func1()
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:85 +0x5c
13:52:31     created by github.com/cockroachdb/pebble.openCleanupManager
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:84 +0x298
13:52:31     
13:52:31     goroutine 88 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c240})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc0006987a8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 89 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c300})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc000698fa8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 90 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c2a0})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc000787fa8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 91 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c360})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc0006f0fa8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 92 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc00065c3c0})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc0006f17a8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 93 [chan receive]:
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x1dd0dc0, 0xc0006bf530})
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:370 +0x9f
13:52:31     runtime/pprof.Do({0x1dd0d50?, 0xc000136000?}, {{0xc000368880?, 0x0?, 0x0?}}, 0xc0006f1fa8)
13:52:31       GOROOT/src/runtime/pprof/runtime.go:44 +0xa3
13:52:31     github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x0?)
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:368 +0x58
13:52:31     created by github.com/cockroachdb/pebble.(*tableCacheShard).init
13:52:31       github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:360 +0xef
13:52:31     
13:52:31     goroutine 835 [select]:
13:52:31     github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
13:52:31       github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:171 +0xdc
13:52:31     created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker
13:52:31       github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:166 +0x5d
13:52:31     
13:52:31     goroutine 832 [select]:
13:52:31     github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFS).startTickerLocked.func1()
13:52:31       github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:566 +0x10e
13:52:31     created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFS).startTickerLocked
13:52:31       github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:555 +0x7a
13:52:31     I231117 13:52:31.357802 1 (gostd) testmain.go:95  [-] 1  Test //pkg/kv/kvserver/raftlog:raftlog_test exited with error code 2
```


</details>

Failed at 57.6k runs

```
ERROR: Test failed, aborting
INFO: Elapsed time: 469.651s, Critical Path: 0.41s
INFO: 57596 processes: 25 internal, 57571 linux-sandbox.
//pkg/kv/kvserver/raftlog:raftlog_test                                NO STATUS

Executed 1 out of 1 test: 1 was skipped.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
INFO: Build Event Protocol files produced successfully.
INFO: Build completed successfully, 57596 total actions
ERROR: exit status 3
```

Could this be an instance of #110028? The test log scope goroutine panics upon finding the config is already active - from pebble table stats `Infof()`. Removing release blocker as a test setup issue.

<details>
<summary>pebble table stats goroutine</summary>

```
I231117 15:18:11.782279 105 3@pebble/event.go:743  [s?,pebble] 1  [JOB 4] all initial table stats loaded
panic: logging already active; first use:
goroutine 105 [running]:
runtime/debug.Stack()
  GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/log.setActive()
  github.com/cockroachdb/cockroach/pkg/util/log/clog.go:437 +0x92
github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(_, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...})
  github.com/cockroachdb/cockroach/pkg/util/log/clog.go:263 +0x50
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1dcca80, 0xc0001b2b70}, 0x4, 0x1, 0x3, 0xa0?, {0x1a6d9a3, 0x2}, {0xc000512010, 0x1, ...})
  github.com/cockroachdb/cockroach/pkg/util/log/channels.go:109 +0x5a5
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
  github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
  github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1dcca80?, 0xc0001b2b70?}, 0x0?}, {0x1a6d9a3?, 0x41eb67?}, {0xc000512010?, 0x17a33a0?, 0x1?})
  github.com/cockroachdb/cockroach/pkg/storage/pebble.go:726 +0x56
github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x0?})
  github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:668 +0x9e
github.com/cockroachdb/pebble.TeeEventListener.func13({0xc0002f8480?})
  github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:743 +0x42
github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc0002f8480)
  github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x322
created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
  github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0xcc
```

</details>

<details><summary>test goroutine</summary>


```
goroutine 114 [running]:
testing.tRunner.func1.2({0x1930a40, 0xc0005101c8})
  GOROOT/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
  GOROOT/src/testing/testing.go:1529 +0x39f
panic({0x1930a40, 0xc0005101c8})
  GOROOT/src/runtime/panic.go:884 +0x213
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.ReportOrPanic({0x1dcca10, 0xc000136000}, 0x0, {0x1aaa851?, 0xc00032c7a0?}, {0xc000512040?, 0x0?, 0x0?})
  github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:417 +0x1b4
github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig({{0xc0002aa800, 0xc00032c620, 0xc00032c640, 0xc00032c798, 0xc00032c65d, {0x1, 0xc0002aa2b0, 0x0, 0xc00032c609, 0xc00032c60a, ...}}, ...})
  github.com/cockroachdb/cockroach/pkg/util/log/flags.go:96 +0x10d
github.com/cockroachdb/cockroach/pkg/util/log.newLogScope.func1(0xc0003a7c70, {0x1dd74c8, 0xc0000076c0}, 0xc0?)
  github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:165 +0x1b8
github.com/cockroachdb/cockroach/pkg/util/log.newLogScope({0x1dd74c8, 0xc0000076c0}, 0xb3?)
  github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:172 +0x42e
github.com/cockroachdb/cockroach/pkg/util/log.ScopeWithoutShowLogs({0x1dd74c8, 0xc0000076c0})
  github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:101 +0x38
github.com/cockroachdb/cockroach/pkg/util/log.Scope({0x1dd74c8, 0xc0000076c0})
  github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:80 +0x4e
github.com/cockroachdb/cockroach/pkg/kv/kvserver/raftlog.TestIterator(0xc0000076c0)
  github.com/cockroachdb/cockroach/pkg/kv/kvserver/raftlog/iter_test.go:169 +0x54
testing.tRunner(0xc0000076c0, 0x1ba1428)
  GOROOT/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
  GOROOT/src/testing/testing.go:1629 +0x3ea
```

</details>

--- a/pkg/kv/kvserver/raftlog/iter_test.go
+++ b/pkg/kv/kvserver/raftlog/iter_test.go
@@ -154,6 +154,7 @@ func TestIteratorEmptyLog(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	eng := storage.NewDefaultInMemForTesting()
+	defer eng.Close()
 	for _, hi := range []kvpb.RaftIndex{0, 1} {
 		it, err := NewIterator(context.Background(), rangeID, eng, IterOptions{Hi: hi})
 		require.NoError(t, err)

